consul/src/main/java/com/linecorp/armeria/internal/consul/ConsulClientBuilder.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.consul;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.SessionProtocol;
+
+public class ConsulClientBuilder {
+    private static final SessionProtocol DEFAULT_CONSUL_PROTOCOL = SessionProtocol.HTTP;
+    private static final String DEFAULT_CONSUL_ADDRESS = "127.0.0.1";
+    private static final int DEFAULT_CONSUL_PORT = 8500;
+    private static final String DEFAULT_CONSUL_API_VERSION = "v1";
+
+    private SessionProtocol consulProtocol = DEFAULT_CONSUL_PROTOCOL;
+    private String consulAddress = DEFAULT_CONSUL_ADDRESS;
+    private int consulPort = DEFAULT_CONSUL_PORT;
+    private String consulApiVersion = DEFAULT_CONSUL_API_VERSION;
+    @Nullable
+    private String consulToken;
+
+    protected ConsulClientBuilder() {
+    }
+
+    protected ConsulClientBuilder consulProtocol(SessionProtocol consulProtocol) {
```

Comment:
Change the modifier to `public` and move the documentation here?
Javadoc is inherited.

Reply:
Oh, sounds good.